### PR TITLE
Update/deployment guides v2

### PR DESCRIPTION
**Pull request by rosieyohannan**
*My last PR got messed up by rebasing :-(*

# Description
Update deployment integration section to put more emphasis on using orbs, and standardise our examples. I recommend [building the site locally](https://github.com/circleci/circleci-docs/blob/master/README-local-development.md) to view changes because some [reusable snippets](https://github.com/circleci/circleci-docs/blob/master/docs/Content-Reuse.md) are used which won't be visible just looking at the diff.

Only the Deployment introduction and examples pages have been edited in this PR - the other pages in the Deployment section were out of scope for this work.

Also - I've put questions and comments in the PR if anyone can help me out with these!

# Reasons
Several customers have commented that the azure section needs more information. Also, having surveyed other cicd providers deployment docs I feel there could be more general deployment guidance before heading straight into examples. Also - orbs are our answer to simplifying deployment so I wanted to make this clearer.

### CODE REVIEW

File: jekyll/_cci2/deployment-examples.md [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+---
+This document presents example config for a variately of popular deployment targets. Many of the examples use orbs: CircleCI and partners have developed a catalogue of orbs that enable you to quickly deploy applications with minimal config. Details of all orbs can be found in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
+
+* TOC
+{:toc}
+
+#### Notes on Examples
+{:.no_toc}
+
+* In order to use orbs you must use `version 2.1` config, and enable pipelines for your project.
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]: I've just merged the pipelines snippet if you want to use that here!

File: jekyll/_cci2/deployment-integrations.adoc [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+These can be added to the project itself (follow links below for detailed instructions), or defined within your configuration. For an example of defining environment variables within your configuration see the non-orb Heroku example below – within the deploy job configuration we have:
++
+```
+environment:
+      HEROKU_APP: "sleepy-refuge-55486"
+```
+** Add project-level <<env-vars#setting-an-environment-variable-in-a-project,environment variables>>
+** Add project-level <<add-ssh-key#,SSH keys>> 
+
+* **Manual Approval**: If your deployment strategy requires a manual approval step, you can include a <<configuration-reference#type,hold/approve>> job within your workflow. A manual approval button will then be available from the workflows map in the CircleCI UI.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-integrations.adoc]: Add in a bullet about using workflows (api?) here, eventually with a link to Michelle's optimization section about workflow/deployment optimization.

File: jekyll/_cci2/deployment-examples.md [hunk 3]
```diff
@@ -0,0 +1,653 @@
...
+
+For detailed information about the AWS S3 orb, refer to the [CircleCI AWS S3 Orb Reference](https://circleci.com/orbs/registry/orb/circleci/aws-s3) page. This section details the use of the AWS S3 orb and `version: 2.1` config for simple deployment, below we will look at the same example without orbs and using using `version: 2` config.
+
+1. For security best practice, create a new [IAM user](https://aws.amazon.com/iam/details/manage-users/) specifically for CircleCI.
+
+2. Add your [AWS access keys](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys) to CircleCI – store your Access Key ID in a variable called `AWS_ACCESS_KEY_ID` and your Secret Access Key in a variable called `AWS_SECRET_ACCESS_KEY`. {% include snippets/env-var-or-context.md %}
+
+3. Use the orb's `sync` command to deploy. Note the use of [workflows]({{ site.baseurl }}/2.0/workflows/) to deploy only if the `build` job passes and the current branch is `master`.
+  
+```
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
```yaml
```

File: jekyll/_cci2/deployment-examples.md [hunk 4]
```diff
@@ -0,0 +1,653 @@
...
+          filters:
+            branches:
+              only: master # Only deploys when the commit is on the Master branch
+```
+
+If pushing to your repo is required, see the [Adding Read/Write Deployment Keys to GitHub or Bitbucket]( {{ site.baseurl }}/2.0/gh-bb-integration/) section of the GitHub and Bitbucket Integration document for instructions. Then, configure the Azure Web App to use your production branch.
+
+## Capistrano
+
+Integrating CircleCI with Capistrano is simple. Once your project is set up to use Capistrano, just run [deployment commands](https://github.com/capistrano/capistrano/blob/master/README.md#command-line-usage) within your CircleCI job steps as required.
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]: Opinionated comment: I would maybe remove the "is simple" line just to be inclusive of user's of all experience levels.

File: jekyll/_cci2/deployment-examples.md [hunk 5]
```diff
@@ -0,0 +1,653 @@
...
+```
+
+### Manual Approval
+{:.no_toc}
+
+For additional control or validation, you can add a manual "hold" step between the dark and live steps as shown in the sample workflow below.
+
+```
+workflows:
+  version: 2 # Not required for `version 2.2` config
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]: 2.2? or 2.1?

File: jekyll/_cci2/deployment-examples.md [hunk 6]
```diff
@@ -0,0 +1,653 @@
...
+      - deploy-job:
+          requires:
+            - build-job
+          filters:
+            branches:
+              only: master
+
+```
+
+If using Google Cloud Functions with Firebase, instruct CircleCI to navigate to the folder where the Google Cloud Functions are held (in this case 'functions') and run npm install by adding the below to `config.yml`:
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
If using Google Cloud Functions with Firebase, instruct CircleCI to navigate to the folder where the Google Cloud Functions are held (in this case 'functions') and run `npm install` by adding the below to `config.yml`:
```

File: jekyll/_cci2/deployment-integrations.adoc [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+== Using Orbs to Simplify Deployment
+
+https://circleci.com/docs/2.0/orb-intro/[Orbs] are available for many common deployment targets to help simplify and streamline your config – check out the full range of available orbs in the https://circleci.com/orbs/registry/[Orbs Registry]. Orbs are packages of reusable configuration, and often, for simple pipelines, simply invoking an orb, including a pre-configured job, and setting any required environment variables/SSH keys within your project will get you the results you need with no further configuration.
```
Comment (teesloane) [inline jekyll/_cci2/deployment-integrations.adoc]: "Orbs are packages of reusable configuration, and often, for simple pipelines, simply invoking an orb, including a pre-configured job, and setting any required environment variables/SSH keys within your project will get you the results you need with no further configuration. "

opinion: I would suggest breaking this into two sentences to reduce the use of commas.

File: jekyll/_cci2/deployment-examples.md [hunk 8]
```diff
@@ -0,0 +1,653 @@
...
+This document presents example config for a variately of popular deployment targets. Many of the examples use orbs: CircleCI and partners have developed a catalogue of orbs that enable you to quickly deploy applications with minimal config. Details of all orbs can be found in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
+
+* TOC
+{:toc}
+
+#### Notes on Examples
+{:.no_toc}
+
+* In order to use orbs you must use `version 2.1` config, and enable pipelines for your project. 
+* If you wish to remain using `version 2.0` config, or are using a self-hosted installation of CircleCI Server, the examples shown here are still relevant because you can view the expanded orb source within the [Orbs Registry](https://circleci.com/orbs/registry/) to see how the jobs are built.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-examples.md]: Not sure this bit about 2.0 users looking at the orb source is legit, but thought I'd stick it in for now as it makes sense to me 🤷🏻‍♀️

File: jekyll/_cci2/deployment-integrations.adoc [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+      - run:
+          name: Run tests
+          command: rake
+
+  deploy:
+    machine:
+        enabled: true
+    working_directory: ~/circleci-demo-workflows
+    environment:
+      HEROKU_APP: "sleepy-refuge-55486" # define env vae $HEROKU_APP
```
Comment (teesloane) [inline jekyll/_cci2/deployment-integrations.adoc]:
```suggestion
      HEROKU_APP: "sleepy-refuge-55486" # define env var $HEROKU_APP
```

File: jekyll/_cci2/deployment-integrations.adoc (hunk 7 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-integrations.adoc]: ha yeah I kept reading that and thinking I need to fix it. So many commas!

File: jekyll/_cci2/deployment-examples.md (hunk 5 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: oops!

File: jekyll/_cci2/deployment-examples.md (hunk 4 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Oh yeah good point!

File: jekyll/_cci2/deployment-examples.md (hunk 1 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: 🎉

Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Just realised I would have to rebase again so I will make this change after this PR gets merged I think 😅

File: jekyll/_cci2/deployment-examples.md [hunk 10]
```diff
@@ -0,0 +1,653 @@
...
+    jobs:
+      - aws-code-deploy/deploy:
+          application-name: myApplication # The name of an AWS CodeDeploy application associated with the applicable IAM user or AWS account.
+          deployment-group: myDeploymentGroup # The name of a new deployment group for the specified application.
+          service-role-arn: myDeploymentGroupRoleARN # The service role for a deployment group.
+          bundle-bucket: myApplicationS3Bucket # The s3 bucket where an application revision will be stored.
+          bundle-key: myS3BucketKey # A key under the s3 bucket where an application revision will be stored.
+```
+
+## Azure Container Registry
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-examples.md]: This whole example is new so needs thoroughly checking

File: jekyll/_cci2/deployment-examples.md [hunk 11]
```diff
@@ -0,0 +1,653 @@
...
+```
+
+### Manual Approval
+{:.no_toc}
+
+For additional control or validation, you can add a manual "hold" step between the dark and live steps as shown in the sample workflow below.
+
+```
+workflows:
+  version: 2 # Not required for `version 2.2` config
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
  version: 2 # Not required for `version 2.1` config
```

File: jekyll/_cci2/deployment-integrations.adoc [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+      HEROKU_APP: "sleepy-refuge-55486"
+```
+** Add project-level <<env-vars#setting-an-environment-variable-in-a-project,environment variables>>
+** Add project-level <<add-ssh-key#,SSH keys>> 
+
+* **Manual Approval**: If your deployment strategy requires a manual approval step, you can include a <<configuration-reference#type,hold/approve>> job within your workflow. A manual approval button will then be available from the workflows map in the CircleCI UI.
+
+== Using Orbs to Simplify Deployment
+
+https://circleci.com/docs/2.0/orb-intro/[Orbs] are available for many common deployment targets to help simplify and streamline your config – check out the full range of available orbs in the https://circleci.com/orbs/registry/[Orbs Registry]. Orbs are packages of reusable configuration, and often, for simple pipelines, simply invoking an orb, including a pre-configured job, and setting any required environment variables/SSH keys within your project will get you the results you need with no further configuration.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-integrations.adoc]:
```suggestion
https://circleci.com/docs/2.0/orb-intro/[Orbs] are available for many common deployment targets to help simplify and streamline your config – check out the full range of available orbs in the https://circleci.com/orbs/registry/[Orbs Registry]. Orbs are packages of reusable configuration. For simple pipelines, simply invoking an orb, including a pre-configured job, and setting any required environment variables/SSH keys within your project will get you the results you need with no further configuration. 
```

File: jekyll/_cci2/deployment-examples.md [hunk 13]
```diff
@@ -0,0 +1,678 @@
...
+            # For example,  hitting a /health endpoint with curl and making sure the dark URL returns a 200.
+```
+
+If you would like more detailed information about various CloudFoundry orb elements that you can use in your configuration workflows, refer to the [CloudFoundry Orb](https://circleci.com/orbs/registry/orb/circleci/cloudfoundry) page in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
+
+### Deploy to Cloud Foundry with 2.0 Config
+
+Cloud Foundry deployments require the Cloud Foundry CLI. Be sure to match the architecture to your Docker image (the commands below assume you are using a Debian-based image). This example pattern implements "Blue-Green" deployments using Cloud Foundry's map-route/unmap-route commands, which is an optional feature above and beyond a basic `cf push`.
+
+#### Install the CLI
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-examples.md]: I would much rather display this as one full config rather than snippet sections - but I don't feel confident to put it together myself. Any help appreciated!

File: jekyll/_cci2/deployment-examples.md [hunk 14]
```diff
@@ -0,0 +1,678 @@
...
+to deploy your application to Heroku, follow the steps below.
+
+### Deploy with the Heroku Orb
+
+1. Create a Heroku account and follow the [Getting Started on Heroku](https://devcenter.heroku.com/start) documentation
+to set up a project in your chosen language.
+
+2. Add the name of your Heroku application and your Heroku API key as environment variables as `HEROKU_APP_NAME` and `HEROKU_API_KEY`, respectively. {% include snippets/env-var-or-context.md %}
+
+3. Use the [Heroku orb](https://circleci.com/orbs/registry/orb/circleci/heroku) to keep your config simple. The `deploy-via-git` installs the Heroku CLI in the primary container, runs any pre deployment steps you define, deploys your application, then runs any post-deployment steps you define. See the Heroku orb page in the [orbs registry](https://circleci.com/orbs/registry/orb/circleci/heroku) for full details of parameters and options:
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-examples.md]: do the pre-deploy and post-deploy steps get added within the workflows sections of the config? There isn 't a usage example for these bits in the orb ref page.

File: jekyll/_cci2/deployment-integrations.adoc [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+//// 
+.Deployment
+image::pipeline-to-deployment.png[Deployment]
+////
+
+== Deployment: The Basics
+
+* **Deployment Job**: To deploy your application, add a <<jobs-steps#jobs-overview,job>> to your `.circleci/config.yml` file and configure the job to run the steps you require. The <<deployment-examples#,deployment examples>> document covers example config for many common deployment targets.
+
+* **Deployment Parameters**: You will need to add environment variables/SSH keys to fulfill your deployment steps and provide autorisation details and parameters.
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-integrations.adoc]: Good spot! Thanks

Comment (rosieyohannan) [inline jekyll/_cci2/deployment-integrations.adoc]:
```suggestion
* **Deployment Parameters**: You will need to add environment variables/SSH keys to fulfill your deployment steps and provide authorization details and parameters.
```

File: jekyll/_cci2/deployment-integrations.adoc [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+          service-role-arn: myDeploymentGroupRoleARN
+          bundle-bucket: myApplicationS3Bucket
+          bundle-key: myS3BucketKey
+```
+
+Under the hood, this orb creates, bundles and deploys your application using your specified parameters set under the `aws-code-deploy/deploy` job declaration.
+
+== A Simple Example Using Heroku
+
+Below is a simple example of deploying a Rails application to Heroku.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-integrations.adoc]:
```suggestion
In this section we will look at a simple example of deploying a Rails application to Heroku. 
```

File: jekyll/_cci2/deployment-examples.md (hunk 8 above)
Comment (teesloane) [reply jekyll/_cci2/deployment-examples.md]: Yep, this is true (and I think you should leave it in)!

File: jekyll/_cci2/deployment-integrations.adoc [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+      HEROKU_APP: "sleepy-refuge-55486"
+```
+** Add project-level <<env-vars#setting-an-environment-variable-in-a-project,environment variables>>
+** Add project-level <<add-ssh-key#,SSH keys>> 
+
+* **Manual Approval**: If your deployment strategy requires a manual approval step, you can include a <<workflows#holding-a-workflow-for-a-manual-approval,hold/approve>> job within your workflow. A manual approval button will then be available from the workflows map in the CircleCI UI.
+
+== Using Orbs to Simplify Deployment
+
+At CircleCI we offer packages of reusable configuration, known as <<orb-intro#,orbs>>. Orbs are available for many common deployment targets, which can help you to simplify and streamline your config. Check out the full range of available orbs in the https://circleci.com/orbs/registry/[Orbs Registry]. For simple pipelines simply invoking an orb, including a pre-configured job, and setting any required environment variables/SSH keys within your project, will get you the results you need with no further configuration.
```
Comment (tay-bird) [inline jekyll/_cci2/deployment-integrations.adoc]: The last sentence here is hard to follow. I think the sentiment you're trying to communicate is, "For simple deployment processes, an orb will get you the results you need with minimal configuration".

File: jekyll/_cci2/deployment-integrations.adoc [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+//// 
+.Deployment
+image::pipeline-to-deployment.png[Deployment]
+////
+
+== Deployment: The Basics
+
+* **Deployment Job**: To deploy your application, add a <<jobs-steps#jobs-overview,job>> to your `.circleci/config.yml` file and configure the job to run the steps you require. The <<deployment-examples#,deployment examples>> document covers example config for many common deployment targets.
+
+* **Deployment Parameters**: You will need to add environment variables/SSH keys to fulfill your deployment steps and provide authorization details and parameters.
```
Comment (tay-bird) [inline jekyll/_cci2/deployment-integrations.adoc]: I had trouble parsing this sentence. I think it'd be clearer in the form, "You may need to add something, something, or something".

File: jekyll/_cci2/deployment-integrations.adoc [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
++
+Environment variables can be added to the project itself (follow links below for detailed instructions), or defined within your configuration. An example of defining environment variables _within_ your configuration can be seen below within the deploy job config for the non-orb Heroku example:
```
Comment (tay-bird) [inline jekyll/_cci2/deployment-integrations.adoc]: Does this come from a larger example? My instinct reading this would be to put `environment` at the root level of my config file, but I'm pretty sure it is actually supposed to be nested. Maybe this could be a link to a section with a larger example, like the ones below?

File: jekyll/_cci2/deployment-integrations.adoc [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+
+== Using Orbs to Simplify Deployment
+
+At CircleCI we offer packages of reusable configuration, known as <<orb-intro#,orbs>>. Orbs are available for many common deployment targets, which can help you to simplify and streamline your config. Check out the full range of available orbs in the https://circleci.com/orbs/registry/[Orbs Registry]. For simple pipelines simply invoking an orb, including a pre-configured job, and setting any required environment variables/SSH keys within your project, will get you the results you need with no further configuration. 
+
+As an example, consider the https://circleci.com/orbs/registry/orb/circleci/aws-code-deploy[AWS CodeDeploy orb]. This orb has a pre-configured `deploy` job, which you can include in your configuration with a single line `aws-code-deploy/deploy` once the orb has been invoked by adding the orbs stanza. For this example deployment can be as simple as:
```
Comment (tay-bird) [inline jekyll/_cci2/deployment-integrations.adoc]: This is a great example.

File: jekyll/_includes/snippets/env-var-or-context.adoc [hunk 21]
```diff
@@ -0,0 +1 @@
+It is best practice to set environment variables at the <<env-vars#setting-an-environment-variable-in-a-project,project level>> or within a <<env-vars#setting-an-environment-variable-in-a-context,context>>.
```
Comment (tay-bird) [inline jekyll/_includes/snippets/env-var-or-context.adoc]: There's a hidden concept here which is that it's a best practice because these are secret keys. Is there a way to say that explicitly in this snippet?

File: jekyll/_cci2/deployment-examples.md [hunk 22]
```diff
@@ -0,0 +1,685 @@
...
+* TOC
+{:toc}
+
+#### Notes on Examples
+{:.no_toc}
+
+* In order to use orbs you must use `version 2.1` config, and enable pipelines for your project. 
+* We have indicated where you need to specify a [docker image for your job]({{ site.baseurl }}/2.0/optimizations/#docker-image-choice) with `<docker-image-name-tag>`.
+* If you wish to remain using `version 2.0` config, or are using a self-hosted installation of CircleCI Server, the examples shown here are still relevant because you can view the expanded orb source within the [Orbs Registry](https://circleci.com/orbs/registry/) to see how the jobs are built.
+* In the examples on this page that use orbs, you will notice that the orbs are versioned with tags, for example, `aws-s3: circleci/aws-s3@x.y.z`. If you copy paste any examples you will need to edit `x.y.z` to specify a version. You can find the available versions listed on the individual orb pages in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
```
Comment (tay-bird) [inline jekyll/_cci2/deployment-examples.md]: Is there a way to always load the latest orb version? Maybe that would be better for the examples?

File: jekyll/_cci2/deployment-examples.md [hunk 23]
```diff
@@ -0,0 +1,685 @@
+---
+layout: classic-docs
+title: "Deployment Examples"
+short-title: "Deployment Examples"
+---
+This document presents example config for a variately of popular deployment targets. Many of the examples use orbs: CircleCI and partners have developed a catalogue of orbs that enable you to quickly deploy applications with minimal config. Details of all orbs can be found in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
+
```
Comment (tay-bird) [inline jekyll/_cci2/deployment-examples.md]: Are there any orbs for deploying Helm packages to Kubernetes? I feel like that'd be a really great example to have.

File: jekyll/_cci2/deployment-examples.md [hunk 24]
```diff
@@ -0,0 +1,685 @@
...
+* If you wish to remain using `version 2.0` config, or are using a self-hosted installation of CircleCI Server, the examples shown here are still relevant because you can view the expanded orb source within the [Orbs Registry](https://circleci.com/orbs/registry/) to see how the jobs are built.
+* In the examples on this page that use orbs, you will notice that the orbs are versioned with tags, for example, `aws-s3: circleci/aws-s3@x.y.z`. If you copy paste any examples you will need to edit `x.y.z` to specify a version. You can find the available versions listed on the individual orb pages in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
+* Any items in these examples that appear within `< >` should be replaced with your own parameters.
+
+## AWS
+
+This section covers deployment to S3, ECR/ECS (Elastic Container Registry/Elastic Container Service), as well as application deployment using AWS Code Deploy. For an in-depth look at deploying to AWS ECS from ECR, see the [Deploying to AWS ECS/ECR document]({{ site.baseurl }}/2.0/ecs-ecr/).
```
Comment (tay-bird) [inline jekyll/_cci2/deployment-examples.md]: All of the deploy examples look really good. I checked the ones I'm familiar with for accuracy. I don't know how anything about Azure, Capistrano, Cloud Foundry, or Firebase.

File: jekyll/_cci2/deployment-integrations.adoc (hunk 17 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-integrations.adoc]: Yes I agree there's no need for that sentence to be so complicated. Fixed.

File: jekyll/_cci2/deployment-integrations.adoc (hunk 19 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-integrations.adoc]: Good point, that is confusing - fixed!

File: jekyll/_includes/snippets/env-var-or-context.adoc (hunk 21 above)
Comment (rosieyohannan) [reply jekyll/_includes/snippets/env-var-or-context.adoc]: I will add a link to https://circleci.com/docs/2.0/contexts/#secrets-masking

File: jekyll/_cci2/deployment-examples.md (hunk 22 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Unfortunately I don't think so

File: jekyll/_cci2/deployment-examples.md (hunk 23 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Oh great, thanks for this idea, I am going to add it to my to do list :-)

File: jekyll/_cci2/deployment-integrations.adoc (hunk 18 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-integrations.adoc]: Thanks, fixed.

File: jekyll/_cci2/deployment-examples.md (hunk 24 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Thanks so much for your help with this :-)

File: jekyll/_cci2/deployment-integrations.adoc [hunk 25]
```diff
@@ -0,0 +1,166 @@
...
+image::pipeline-to-deployment.png[Deployment]
+////
+
+== Deployment: The Basics
+
+* **Deployment Job**: To deploy your application, add a <<jobs-steps#jobs-overview,job>> to your `.circleci/config.yml` file and configure the job to run the steps you require. The <<deployment-examples#,deployment examples>> document covers example config for many common deployment targets.
+
+* **Deployment Parameters**: You may need to add environment variables and SSH keys to fulfill your deployment steps.
++
+Environment variables can be added to the project itself via the CircleCI UI (follow links below for detailed instructions), or defined within your configuration. An example of defining environment variables _within_ your configuration can be seen <<#2.0-config,further down this page within the deploy job config for the non-orb Heroku example>>.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-integrations.adoc]:
```suggestion
Environment variables can be added to the project itself via the CircleCI UI (follow links below for detailed instructions), or defined within your configuration. An example of defining environment variables _within_ your configuration can be seen <<2.0-config,further down this page within the deploy job config for the non-orb Heroku example>>.
```

File: jekyll/_cci2/deployment-integrations.adoc [hunk 26]
```diff
@@ -0,0 +1,166 @@
...
+image::pipeline-to-deployment.png[Deployment]
+////
+
+== Deployment: The Basics
+
+* **Deployment Job**: To deploy your application, add a <<jobs-steps#jobs-overview,job>> to your `.circleci/config.yml` file and configure the job to run the steps you require. The <<deployment-examples#,deployment examples>> document covers example config for many common deployment targets.
+
+* **Deployment Parameters**: You may need to add environment variables and SSH keys to fulfill your deployment steps.
++
+Environment variables can be added to the project itself via the CircleCI UI (follow links below for detailed instructions), or defined within your configuration. An example of defining environment variables _within_ your configuration can be seen <<2.0-config,further down this page within the deploy job config for the non-orb Heroku example>>.
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-integrations.adoc]:
```suggestion
Environment variables can be added to the project itself via the CircleCI UI (follow links below for detailed instructions), or defined within your configuration. An example of defining environment variables _within_ your configuration can be seen further down this page within the deploy job config for the non-orb Heroku example.
```

File: jekyll/_cci2/deployment-examples.md [hunk 27]
```diff
@@ -0,0 +1,685 @@
...
+firebase login:ci
+```
+
+Add the generated token to the CircleCI project's environment variables as `$FIREBASE_DEPLOY_TOKEN`. {% include snippets/env-var-or-context.md %}
+
+The following example shows how you can add a deploy to Firebase job to your project's `config.yml` file. This snippet assumes you already have a job to build your application, called `build-job`, and introduces a deployment workflow that only runs the deployment job once the build job has completed **and** you're on the master branch.
+
+```
+
+ deploy-job:
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
  deploy-job:
```
Should be two spaces of indentation for correct yaml.

Comment (teesloane) [reply jekyll/_cci2/deployment-examples.md]: Actually, I think everything in this block might need one additional space. also, this example is pretty close to being a full config, no? Maybe we should just add the full thing for increased copy-paste-ability (I think it just needs a `version` and `jobs` key

File: jekyll/_cci2/deployment-examples.md [hunk 28]
```diff
@@ -0,0 +1,685 @@
...
+
+```
+firebase login:ci
+```
+
+Add the generated token to the CircleCI project's environment variables as `$FIREBASE_DEPLOY_TOKEN`. {% include snippets/env-var-or-context.md %}
+
+The following example shows how you can add a deploy to Firebase job to your project's `config.yml` file. This snippet assumes you already have a job to build your application, called `build-job`, and introduces a deployment workflow that only runs the deployment job once the build job has completed **and** you're on the master branch.
+
+```
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
```yaml
```

File: jekyll/_cci2/deployment-examples.md [hunk 29]
```diff
@@ -0,0 +1,685 @@
...
+## Google Cloud Platform
+
+Before deploying to Google Cloud Platform, you will need to authorize the Google Cloud SDK and set default configuration settings. Refer to the [Authorizing the Google Cloud SDK]({{ site.baseurl }}/2.0/google-auth/) document for full details.
+
+### Using Google Cloud Orbs
+{:.no_toc}
+
+There are several Google Cloud orbs available in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/) that you can use to simplify your deployments. For example, the [Google Kubernetes Engine (GKE) orb](https://circleci.com/orbs/registry/orb/circleci/gcp-gke#usage-publish-and-rollout-image) has a pre-built job to build and publish a Docker image, and roll the image out to a GKE cluster, as follows:
+
+```
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
```yaml
```

File: jekyll/_cci2/deployment-examples.md (hunk 13 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Yeah just show a full config example and talk about it - I think it's easier to follow that way but that's just me, I am happy to be wrong :-)

File: jekyll/_cci2/deployment-examples.md [hunk 30]
```diff
@@ -0,0 +1,685 @@
+---
+layout: classic-docs
+title: "Deployment Examples"
+short-title: "Deployment Examples"
+---
+This document presents example config for a variately of popular deployment targets. Many of the examples use orbs: CircleCI and partners have developed a catalogue of orbs that enable you to quickly deploy applications with minimal config. Details of all orbs can be found in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: I think I disagree, I think colons can be used when you have two clauses, when the second expands on the first...

File: jekyll/_cci2/deployment-examples.md [hunk 31]
```diff
@@ -0,0 +1,685 @@
...
+                  only: master # Only deploy when the commit is on the Master branch
+
+    jobs: # Define the build and deploy jobs
+      build:
+        docker: # Use the Docker executor for the build job
+          - image: <image-name-and-tag> # Specify the Docker image to use for the build job
+      ... # build job steps omitted for brevity
+      deploy:
+        docker: # Use the Docker executor for the deploy job
+          - image: <image-name-and-tag>  Specify the Docker image to use for the deploy job
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Good spot!

File: jekyll/_cci2/deployment-examples.md [hunk 32]
```diff
@@ -0,0 +1,685 @@
...
+      cf -v
+      cf api https://api.run.pivotal.io  # alternately target your private Cloud Foundry deployment
+      cf auth "$CF_USER" "$CF_PASSWORD"
+      cf target -o "$CF_ORG" -s "$CF_SPACE"
+```
+
+#### Dark Deployment
+{:.no_toc}
+
+This is the first step in a Blue-Green deployment, pushing the application to non-production routes.
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Yeah good point - I asked on Slack to find out I think.

File: jekyll/_cci2/deployment-examples.md (hunk 27 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: Yeah that's a good idea, I'll see what I can do

File: jekyll/_cci2/deployment-examples.md [hunk 33]
```diff
@@ -0,0 +1,751 @@
...
+
+This is the first step in a [Blue-Green](https://docs.cloudfoundry.org/devguide/deploy-apps/blue-green.html) deployment, pushing the application to non-production routes.
+
+{% raw %}
+
+```yaml
+      - run:
+          name: CF Deploy
+          command: |
+            # push artifacts on "dark" subdomain, not yet starting so we can attach environment variables
```
Comment (rosieyohannan) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
            # push artifacts on "dark" subdomain, and set environment variables before running `cf start`.
```

File: jekyll/_cci2/deployment-examples.md (hunk 27 above)
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: I'm going to add this to my to do list - I think Adam is working on a new Firebase example anyway.

File: jekyll/_cci2/deployment-examples.md (hunk 30 above)
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/deployment-examples.md]: But to make this easier I'm just going to use a `.`

File: jekyll/_cci2/deployment-examples.md [hunk 34]
```diff
@@ -0,0 +1,751 @@
...
+    jobs:
+      - aws-code-deploy/deploy:
+          application-name: <my-application> # The name of an AWS CodeDeploy application associated with the applicable IAM user or AWS account.
+          deployment-group: <my-deployment-group> # The name of a new deployment group for the specified application.
+          service-role-arn: <my-deployment-group-role-ARN> # The service role for a deployment group.
+          bundle-bucket: <my-application-S3-bucket> # The s3 bucket where an application revision will be stored.
+          bundle-key: <my-S3-bucket-key> # A key under the s3 bucket where an application revision will be stored.
+```
+
+{% raw %}
```
Comment (teesloane) [inline jekyll/_cci2/deployment-examples.md]:
```suggestion
{% endraw %}
```
